kie-internal/src/main/java/org/kie/internal/utils/NotificationPublisher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.internal.utils;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.kie.api.task.model.OrganizationalEntity;
+
+/**
+ * Responsible for providing various types of publishers of notifications 
+ * e.g. email, sms, message, etc.
+ */
+public interface NotificationPublisher {
+
+    /**
+     * Publishes given notification based on the actual implementation.
+     * This method should never thrown exceptions as they might be interfering with processing logic. 
+     * 
+     * @param sender identifier of the sender to be attached to the message produced from the notification
+     * @param subject subject to be used when publishing the notification
+     * @param recipients list of users or groups the notification should be send to
+     * @param body actual message body
+     */
+    void publish(String sender, String subject, Set<OrganizationalEntity> recipients, String body);
```

Comment:
Just an idea, could it be beneficial to have the sender also OrganizationalEntity? Various publishers could use different pieces of information about the sender.

Reply:
I intentionally left it as string as it might represent not really user/group or any other org entity and thus is should be up to publisher impl to decide how to deal with it.